pypesto/hierarchical/optimal_scaling_approach/optimal_scaling_parameter.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+from ...C import InnerParameterType
+from ..parameter import InnerParameter
+
+logger = logging.getLogger(__name__)
+
+
+class OptimalScalingParameter(InnerParameter):
+    """Inner parameter of the optimal scaling hierarchical optimization problem.
+
+    Attributes
+    ----------
+    category:
+        Category index.
+    group:
+        Group index.
+    value:
+        Current value of the inner parameter.
+    estimate:
+        Whether to estimate inner parameter in inner subproblem.
+    """
+
+    def __init__(
+        self,
+        *args,
+        category: int = None,
+        group: int = None,
+        estimate: bool = False,
+        **kwargs,
+    ):
+        """Construct.
+
+        Parameters
+        ----------
+        See class attributes.
+        """
+        super().__init__(*args, **kwargs)
+        if self.inner_parameter_type != InnerParameterType.OPTIMALSCALING:
+            raise ValueError(
+                "For the OptimalScalingParameter class, the parameter type has to be optimal_scaling."
+            )
+
+        if group is None:
```

Comment:
does the constructor need None default arguments? If not, please remove the default args there (can still keep this check to be sure).